lib/src/ruby/module.cc

```diff
@@ -953,20 +958,19 @@ namespace facter { namespace ruby {
 
         // Look for "facter" subdirectories on the load path
         for (auto const& directory : ruby.get_load_path()) {
-            // Get the absolute directory name
             boost::system::error_code ec;
-            path dir = absolute(directory);
+            // Use forward-slash to keep this consistent with Ruby conventions.
+            auto dir = canonicalize(directory) + "/facter";
```

Comment:
I can try to add a test that adds a Ruby LOAD_PATH for a fixture, then also sets it as a search path, and make sure the custom fact is only loaded once (somehow). Detecting that is the slightly tricky part.